sdk/test-utils/recorder/README.md

```diff
@@ -1,9 +1,450 @@
+  store the response received from the live resource that was originally
+  targeted. We use a couple of mechanisms to do this for NodeJS and the
+  browser. The output files are stored in `recordings/node/*` and in
+  `recordings/browser/*`, which are relative to the root of the project you're
+  be working on.
+- To **playback** means to intercept any HTTP request and to respond it with the
+  stored response of a previously recorded matching request.
+
+## Getting started
+
+We're about to go through how to set up your project to use the test-utils-recorder.
+We'll be using git and [rush](https://rushjs.io). We understand that both tools
+can be challenging to use, but we expect you to know how to use them by the
+time you decide to use this package. If you encounter any problem with these
+tools, please make sure to check at their documentation first, but we're also
+happy to answer any question! We appreciate that you're reading this 💙
+
+Keep in mind that test-utils-recorder is not a published package. It is only intended to be used
+by our libraries (at least for now).
+
+### Installing the package
+
+To install the test-utils-recorder package, you'll need to start by cloning our
+azure-sdk-for-js repository. One way of doing this is by using the git command
+line interface, as follows:
+
+    cd /path/to/my/github/repositories
+    git clone https://github.com/Azure/azure-sdk-for-js/
+
+Having cloned this repository, let's set it up by running the following rush commands:
+
+    cd azure-sdk-for-js
+    rush update
+    rush install
+    rush build
+
+This will optimistically assume you're in a fresh clone.
+
+From this point forward, we'll assume that you're developing (perhaps
+contributing!) to one of our libraries. So, your next step is to change
+directory to the path relevant to your project. Let's say you want to add
+test-utils-recorder to the package `@azure/keyvault-keys` (it already uses test-utils-recorder,
+but bear with us), you'll be doing the following:
+
+    cd sdk/keyvault/keyvault-keys
+
+One there, you can add the test-utils-recorder package by running the following rush command:
+
+    rush add -p @azure/test-utils-recorder
+
+And you're ready! Now you can use the common recorder in your code, as shown below:
+
+```typescript
+import * as commonRecorder from "@azure/test-utils-recorder";
+```
+
+The common recorder provides the following public methods and properties:
+
+- `record`: Which deals with recording and playing back the network requests,
+  depending on the value assigned to the `TEST_MODE` environment variable.
+  If `TEST_MODE` equals to `record`, it will automatically store every network
+  request in a plain text file in the folder `recordings` at the root of your
+  repository (which for our example case is the root of the
+  `@azure/keyvault/keyvault-keys` repository). It must be used inside of Mocha's
+  tests, since it uses [nise](https://www.npmjs.com/package/nise) under the hood.
```

Comment:
No. 
For node tests, we leverage nock library to intercept and record Node's http.request.
For browser tests, we leverage nise library to intercept and record browser's XMLHttpRequest.

The reason for `It must be used inside of Mocha's tests` is because the recorder package is only configured for the mocha tests as recommended in this repo.

Reply:
We can say, the recorder package assumes that the tests in the library are leveraging [mocha](https://mochajs.org/) and [rollup](https://rollupjs.org/guide/en/) (and [karma](https://karma-runner.github.io/latest/index.html) test runner for browser tests) as suggested by the [template](https://github.com/Azure/azure-sdk-for-js/tree/master/sdk/template/template) package in the repo.

Reply:
You may use this statement. 
https://github.com/Azure/azure-sdk-for-js/blame/master/sdk/test-utils/recorder/GUIDELINES.md#L20